SalesforceCommerceCloud/pwa-kit
# Fix localstorage bug

There was a bug with the `useCustomerId` hook. The issue is that the hook was never triggered on localstorage changes on the same document. It works if you modify the value thru chrome devtools, but it doesn't work when a change is triggered by javascript on the page.

We had:

```
window.addEventListener('storage', handleStorageChange)
```

But according to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event) and stackoverflow posts, it appears this event is only triggered when the change happens on another document.

> The storage event of the [Window](https://developer.mozilla.org/en-US/docs/Web/API/Window) interface fires when a storage area (localStorage) has been modified in the context of another document.

> Note: This won't work on the same page that is making the changes — it is really a way for other pages on the domain using the storage to sync any changes that are made. Pages on other domains can't access the same storage objects.

Unfortunately our changes happens on the same document, so the fix is to manually trigger a storage update event.

### How to test?

- pull code and build sdk
- cd packages/test-commerce-sdk-react
- go to localhost:3000/customerId
- you should see customer id being displayed on the screen
